src/serverrpc.cpp

```diff
@@ -0,0 +1,151 @@
+/******************************************************************************\
+ * Copyright (c) 2021
+ *
+ * Author(s):
+ *  dtinth
+ *
+ ******************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ *
+ \******************************************************************************/
+
+#include "serverrpc.h"
+
+CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* parent ) : QObject ( parent )
+{
+    pRpcServer->HandleMethod ( "jamulus/getMode", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "mode", "server" } };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/getServerStatus", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonArray            clients;
+        CVector<CHostAddress> vecHostAddresses;
+        CVector<QString>      vecsName;
+        CVector<int>          veciJitBufNumFrames;
+        CVector<int>          veciNetwFrameSizeFact;
+
+        pServer->GetConCliParam ( vecHostAddresses, vecsName, veciJitBufNumFrames, veciNetwFrameSizeFact );
+
+        // we assume that all vectors have the same length
+        const int iNumChannels = vecHostAddresses.Size();
+
+        // fill list with connected clients
+        for ( int i = 0; i < iNumChannels; i++ )
+        {
+            if ( vecHostAddresses[i].InetAddr == QHostAddress ( static_cast<quint32> ( 0 ) ) )
+            {
+                continue;
+            }
+            QJsonObject client{
+                { "address", vecHostAddresses[i].toString ( CHostAddress::SM_IP_PORT ) },
+                { "name", vecsName[i] },
+                { "jitterBufferSize", veciJitBufNumFrames[i] },
+                { "channels", pServer->GetClientNumAudioChannels ( i ) },
+            };
```

Comment:
We do not expose the client ID here, but it would be useful to do so. I'm unsure whether to expose it as a `id` key or whether to change `clients` to a `QJsonObject` so that we an use the ID as an index. Hm...

Reply:
Implemented as `"id"` field similar to `jamulusclient/clientListReceived`.